cl/users/tasks.py

```diff
@@ -102,3 +106,35 @@ def update_mailchimp(self, email, status):
             email,
             r.status_code,
         )
+
+
+@app.task(bind=True, max_retries=3, interval_start=5 * 60)
+def process_retry_email(
+    self: Task,
+    failed_pk: int,
+) -> int:
+    """Task to retry failed email messages"""
+
+    failed_email = FailedEmail.objects.get(pk=failed_pk)
+    if failed_email.status != STATUS_TYPES.SUCCESSFUL:
+        # Only execute this task if it has not been previously processed.
+        failed_email.status = STATUS_TYPES.IN_PROGRESS
+        failed_email.save()
+        # Compose email from stored message.
+        email = failed_email.stored_email.convert_to_email_multipart()
+        try:
+            email.send()
+        except Exception as exc:
+            # In case of error when sending e.g: SES downtime, retry the task.
+            raise self.retry(exc=exc)
+        failed_email.status = STATUS_TYPES.SUCCESSFUL
+        failed_email.save()
+
+
+def send_failed_email(
+    failed_pk: int,
+    start: datetime,
+) -> None:
+    return chain(
+        process_retry_email.si(failed_pk),
+    ).apply_async(eta=start)
```

Comment:
I think there's a better way to do this without using a chain. Mind digging in celery a bit? I think you can probably eliminate this function.

Reply:
Changed, removed this function, and called the task directly.